src/api/form/PDFForm.ts

```diff
@@ -499,6 +508,68 @@ export default class PDFForm {
     return PDFTextField.of(text, text.ref, this.doc);
   }
 
+  /**
+   * Flatten all form fields.
+   * 
+   * Flattening a form field will take the current appearance and make that part
+   * of the pages content stream. All form fields and annotations associated are removed.
+   * 
+   * For example:
+   * ```js
+   * const form = pdfDoc.getForm();
+   * form.flatten();
+   * ```
+   */
+  flatten() {
+    this.updateFieldAppearances();
+
+    const fields = this.getFields();
+    const pages = this.doc.getPages();
+
+    for (const field of fields) {
```

Comment:
This syntax is nice and I find it much more readable than low-level indexing loops. However, the decision to use optimized index-loops in the pdf-lib codebase has already been made (for performance reasons). So please convert this loop to use the following syntax: https://github.com/Hopding/pdf-lib/blob/b169e2fda05beacfe722ed1b3e73e4ec9a59a589/src/api/PDFDocument.ts#L1168